sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/models/FeatureFlagConfigurationSetting.java

```diff
@@ -208,6 +265,16 @@ public FeatureFlagConfigurationSetting setClientFilters(List<FeatureFlagFilter>
      */
     public FeatureFlagConfigurationSetting addClientFilter(FeatureFlagFilter clientFilter) {
         clientFilters.add(clientFilter);
+        updateSettingValue();
         return this;
     }
+
+    private void updateSettingValue() {
+        try {
+            super.setValue(writeFeatureFlagConfigurationSetting(this));
+        } catch (IOException exception) {
+            LOGGER.logExceptionAsError(new IllegalArgumentException(
+                "Can't parse Feature Flag configuration setting value. Exception:" + exception));
```

Comment:
Don't use `+ exception`. Instead pass it as param `LOGGER.logExceptionAsError(new IllegalArgumentException(
                "Can't parse Feature Flag configuration setting value.", exception))`. Or use `exception.getMessage()`.